### PR TITLE
Feature/create a note

### PR DESCRIPTION
**Pull request by marijan25**
Create pull request for the branch feature/create-a-note
Form modal

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,21 +5,27 @@
   "dependencies": {
```
Comment (devKid88) [inline package.json:5]: for this ticket we should not have changes in the package.json

File: package.json [hunk 2]
```diff
@@ -5,21 +5,27 @@
...
+    "styled-components": "^5.3.3",
     "web-vitals": "^2.1.4"
   },
   "scripts": {
     "start": "react-scripts start",
     "build": "react-scripts build",
     "test": "react-scripts test",
-    "eject": "react-scripts eject"
+    "eject": "react-scripts eject",
+    "server": "json-server --watch data.json --port 8000"
```
Comment (devKid88) [inline package.json]: remove json server

File: src/App.js [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
+      const notesFromServer = await fetchNotes()
+      setNotes(notesFromServer)
+    }
+    getNotes()
+  }, [])
+  const fetchNotes = async () => {
+    const res = await fetch('http://localhost:8000/notes')
+    const data = await res.json()
+    return data
+  }
```
Comment (devKid88) [inline src/App.js]: You can remove all API calls, this is not in the scope of ticket or project

File: src/components/Modal.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+                autoFocus>
+                  Cancel
+              </Button>
+            </Stack>
+          </form>
+        </DialogContent>
+      </BootstrapDialog>
+    </Grid>
+  );
+}
```
Comment (devKid88) [inline src/components/Modal.js:24]: each react component should be in its own file, please create a file for each component, then I will review this logic

**Review by devKid88 — CHANGES_REQUESTED** [review]

File: src/App.js [hunk 5]
```diff
@@ -1,28 +1,69 @@
...
+    {
+      id: 4,
+      title: "Edit note",
+      content: "When a user clicks on edit, the same modal as Add note should be displayed with the same rules with few changes - Instead of Add Button, there should be an Edit button."
+    },
+    {
+      id: 5,
+      title: "Delete note",
+      content: "When a user clicks on delete button, confirmation modal should be displayed with a question Delete note title* ?"
+    }
```
Comment (devKid88) [inline src/App.js]: move this to the separated const and use it here

File: src/components/AddForm.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
```
Comment (devKid88) [inline src/components/AddForm.js]: remove

File: src/components/AddForm.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+  const [titleNote, setTitleNote] = useState("");
+  const [contentNote, setContentNote] = useState("");
+  const handleTitleChange = (event) => {
+    setTitleNote(event.target.value);
+  };
+  const handleContentChange = (event) =>{
+    setContentNote(event.target.value);
+  };
+  const [title, setTitle] = useState('');
+  const [content, setContent] = useState('');
```
Comment (devKid88) [inline src/components/AddForm.js]: Can you explain to me why we have titleNote and title, and contentNote and content?

File: src/components/Modal.js (hunk 4 above)
Comment (devKid88) [reply src/components/Modal.js:24]: still valid

**Review by devKid88 — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (devKid88) [reply package.json:5]: still valid

File: src/components/AddForm.js (hunk 7 above)
Comment (marijan25) [reply src/components/AddForm.js]: TitleNote and ContentNote are not required in the code, so I will delete them. Title and content disable the add button if the required fields are not filled

**Review by devKid88 — APPROVED** [review]

**Pull request merged**